test/rnn_seq_api.hpp

```diff
@@ -0,0 +1,1663 @@
+        case miopenGRU: std::cout << " gru "; break;
+        default: break;
+        }
+
+        auto& inLens = input.desc.GetLengths();
+        auto& hLens  = xHiddenState.desc.GetLengths();
+
+        std::cout << " --batch_size " << inLens[0] << " --seq_len " << inLens[1] << " --in_vec "
+                  << inLens[2] << " --hid_h " << hLens[2] << " --num_layer " << rnnDesc.nLayers
+                  << " -r " << rnnDesc.dirMode << " -b " << rnnDesc.biasMode << " -p "
+                  << rnnDesc.inputMode << " -a " << rnnDesc.algoMode;
+
+        bool useDropout = !miopen::float_equal(miopen::deref(rnnDesc.dropoutDesc).dropout, 0);
+
+        std::cout << " --io_layout "
+                  << miopen::RNNDescriptor::getBaseLayoutFromDataTensor(input.desc)
+                  << " --use_dropout " << useDropout;
+        if(useDropout)
+            std::cout << " --dropout " << miopen::deref(rnnDesc.dropoutDesc).dropout;
+
+        auto& samplesLen = input.desc.GetSequenceLengthsVector();
+        std::cout << " --seq_len_array ";
+        for(int i = 0; i < inLens[0]; i++)
+        {
+            if(i < inLens[0] - 1)
+            {
+                std::cout << samplesLen.at(i) << ",";
+            }
+            else
+            {
+                std::cout << samplesLen.at(i);
+            }
+        }
+        std::cout << std::endl;
+
+        if(badtensor >= 0)
+        {
+            if(badtensor < 3)
+            {
+                std::cout << "FWD Train LSTM: " << std::endl;
+                switch(badtensor)
+                {
+                case(0):
+                    std::cout << "Output tensor output failed verification." << std::endl;
+                    break;
+                case(1):
+                    std::cout << "Hidden state tensor failed verification." << std::endl;
+                    break;
+                case(2): std::cout << "Cell state tensor failed verification." << std::endl; break;
+                default: break;
+                }
+            }
+            else if(badtensor < 6)
+            {
+                std::cout << "BWD Train LSTM: " << std::endl;
+                switch(badtensor)
+                {
+                case(3):
+                    std::cout << "Output tensor output failed verification." << std::endl;
+                    break;
+                case(4):
+                    std::cout << "Hidden state tensor failed verification." << std::endl;
+                    break;
+                case(5): std::cout << "Cell state tensor failed verification." << std::endl; break;
+                default: break;
+                }
+            }
+            else if(badtensor == 6)
+            {
+                std::cout << "WRW Train LSTM " << std::endl;
+            }
+        }
+    }
+};
+
+//****************************************************
+// RNN TRAIN
+//****************************************************
+
+template <class T>
+struct rnn_ref
+{
+    struct FWDResObj
```

Comment:
[highly recommended] Can we use CamelCase in the new code? FwdSomething, BwdSomething, WrwSomething...

Reply:
Ok. Also FwData, BwData, BwFilter of BwWeights

Reply:
[resolved]